torchtune/models/llama3/_parallelism.py

```diff
@@ -10,26 +10,43 @@
 from torch.distributed.tensor.parallel import ColwiseParallel, RowwiseParallel
 from torch.distributed.tensor.parallel.style import ParallelStyle
 
+from torchao.float8.float8_tensor_parallel import (
+    Float8ColwiseParallel,
+    Float8RowwiseParallel,
+)
 
-# Define the Tensor Parallel plan for Llama3 model, which will also be shared with 3.1, 3.2, and 3.3 models
-BASE_LLAMA_TP_PLAN = {
-    "tok_embeddings": RowwiseParallel(input_layouts=Replicate()),
-    "output": ColwiseParallel(output_layouts=Replicate()),
-    "layers.*.attn.q_proj": ColwiseParallel(),
-    "layers.*.attn.k_proj": ColwiseParallel(),
-    "layers.*.attn.v_proj": ColwiseParallel(),
-    "layers.*.attn.output_proj": RowwiseParallel(),
-    "layers.*.mlp.w1": ColwiseParallel(),
-    "layers.*.mlp.w2": RowwiseParallel(),
-    "layers.*.mlp.w3": ColwiseParallel(),
-}
-
-
-def base_llama_tp_plan() -> Dict[str, ParallelStyle]:
+
+def base_llama_tp_plan(enable_float8: bool) -> Dict[str, ParallelStyle]:
     """
     Helper function to get the base tensor parallel plan for Llama3 model, which will also be shared with 3.1, 3.2, and 3.3 models
 
+    Args:
+        enable_float8 (bool): Whether float8 training is enabled and TP classes should use fp8 alternatives
+
     Returns:
         Dict[str, Any]: The tensor parallel plan for Llama3 model.
     """
-    return BASE_LLAMA_TP_PLAN
+    if enable_float8:
+        rowwise_parallel, colwise_parallel = (
+            Float8RowwiseParallel,
```

Comment:
these modules are specific to tensorwise scaling, so if in the future torchtune wants to enable more recipes (such as rowwise scaling) there will have to be additional gating on this